core/app/models/spree/zone.rb

```diff
@@ -171,6 +171,21 @@ def contains?(target)
       true
     end
 
+    def state_list
+      case kind
+      when 'country'
+        zoneables.map(&:states)
+      when 'state'
+        zoneables
+      else
+        []
```

Comment:
I'm wondering, if we need this checking here in case where we have only two things, `state` and `country` to take into consideration, what is purpose of this check ?

Reply:
I based on the `Spree::Zone#contry_list` ([source](https://github.com/spree/spree/blob/master/core/app/models/spree/zone.rb#L118)). 

In fact, there is no not null db constraint on the column 
https://github.com/spree/spree/blob/master/core/db/migrate/20140911173301_add_kind_to_zone.rb#L3

Reply:
saw that method, it's pretty old, was wondering why we have to check other option if we have only two option to choose during create new zone 🤔

Reply:
That's right.  Though, having in mind that the db allows null kinds maybe it's safer to respond with empty array for such edge cases. If we just get rid of it now some people may suffer after the spree bump. WDYT? @przemosk

Reply:
this is one way, another one what I would like to suggest is make this more consistent ( atm in backend new zone form we have pre-selected `state` based option ) and avoid repeat those habit ( like in country method ) where we checking condition which don't have chance to exists. 

We could resolve that by add some default option in this migration what you bring up in your above comment, for instance.

Reply:
I'll try to prepare a commit for that

Reply:
@przemosk https://github.com/spree/spree/pull/10589/commits/a8fc7ae89d9725d8f3abfbfb779ec71c90fafb40